benchmarks/Kestrel.Performance/InMemoryTransportBenchmark.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    public class InMemoryTransportBenchmark
+    {
+        private const int _plaintextExpectedResponseLength = 132;
+        private const int _plaintextPipelinedExpectedResponseLength = _plaintextExpectedResponseLength * RequestParsingData.Pipelining;
+
+        private IWebHost _host;
+        private InMemoryConnection _connection;
+
+        [GlobalSetup(Target = nameof(Plaintext) + "," + nameof(PlaintextPipelined))]
+        public void GlobalSetupPlaintext()
+        {
+            var transportFactory = new InMemoryTransportFactory(connectionsPerEndPoint: 1);
+
+            _host = new WebHostBuilder()
+                // Prevent VS from attaching to hosting startup which could impact results
+                .UseSetting("preventHostingStartup", "true")
+                .UseKestrel()
+                // Bind to a single non-HTTPS endpoint
+                .UseUrls("http://127.0.0.1:5000")
+                .ConfigureServices(services => services.AddSingleton<ITransportFactory>(transportFactory))
+                .Configure(app => app.UseMiddleware<PlaintextMiddleware>())
+                .Build();
+
+            _host.Start();
+
+            // Ensure there is a single endpoint and single connection
+            _connection = transportFactory.Connections.Values.Single().Single();
+        }
+
+        [GlobalCleanup]
+        public void GlobalCleanup()
+        {
+            _host.Dispose();
+        }
+
+        [Benchmark]
+        public void Plaintext()
+        {
+            _connection.SendRequestAsync(RequestParsingData.PlaintextTechEmpowerRequest).Wait();
+            _connection.GetResponseAsync(_plaintextExpectedResponseLength).Wait();
+        }
```

Comment:
Why not async?
```csharp
public async Task Plaintext()
{
    await _connection.SendRequestAsync(RequestParsingData.PlaintextTechEmpowerRequest);
    await _connection.GetResponseAsync(_plaintextExpectedResponseLength);
}
```

Reply:
Results are about the same so I'll make this change.  I originally made the method sync to match the `RequestParsing` benchmark methods.

# Sync
Method |      Mean |     Error |    StdDev |      Op/s | Allocated |
------------------- |----------:|----------:|----------:|----------:|----------:|
Plaintext | 11.262 us | 0.5450 us | 0.8157 us |  88,795.7 |     247 B |
PlaintextPipelined |  7.509 us | 0.4845 us | 0.7252 us | 133,177.1 |      21 B |

# Async
Method |      Mean |     Error |    StdDev |      Op/s | Allocated |
------------------- |----------:|----------:|----------:|----------:|----------:|
Plaintext | 11.744 us | 0.5272 us | 0.7890 us |  85,148.1 |     372 B |
PlaintextPipelined |  7.107 us | 0.2767 us | 0.4142 us | 140,696.9 |      30 B |